src/common/snippets/src/pass/common_optimizations.cpp

```diff
@@ -19,21 +19,35 @@ namespace ov {
 namespace snippets {
 namespace pass {
 
-bool CommonOptimizations::CanOptimizeParallelWA(const std::shared_ptr<const ov::Node>& node, size_t minimal_concurrency) {
-    if (!ov::is_type<ov::op::v0::MatMul>(node))
+namespace {
+size_t get_lcm(size_t a, size_t b) {
+    std::function<size_t(size_t, size_t)> get_gcd;
+    get_gcd = [&get_gcd](size_t a, size_t b) {
+        if (b == 0)
+            return a;
+        return get_gcd(b, a % b);
+    };
+    return a / get_gcd(a, b) * b;
+}
+
+bool is_supported_matmul_for_split_dim_m_optimization(const std::shared_ptr<const ov::Node>& node) {
+    const auto matmul = ov::as_type_ptr<const ov::op::v0::MatMul>(node);
+    return matmul && !matmul->get_transpose_a() && !matmul->is_dynamic() && node->get_shape().size() == 3; // It's needed only for 3D MHA patterns
+}
+}  // namespace
+
+bool CommonOptimizations::CanOptimizeParallelWA(const std::shared_ptr<const ov::Node>& node, size_t concurrency) {
+    if (!is_supported_matmul_for_split_dim_m_optimization(node))
         return false;
-    // It's needed only for 3D MHA patterns
     const auto mm_shape = node->get_shape();
-    if (mm_shape.size() != 3)
-        return false;
     const auto current_parallel_work_amount =
         std::accumulate(mm_shape.rbegin() + 2, mm_shape.rend(), size_t(1), std::multiplies<size_t>());
```

Comment:
Minor: looks like an unnecessary complication. We should either rely on the checks inside `is_supported_matmul_for_split_dim_m_optimization` and assume the shape rank to be 3, then we can just take a single dim value. Alternatively, we can write a more general code, so the checks in `is_supported_matmul_for_split_dim_m_optimization` could be updated independently. In this case, we need to check that the shape rank>=2.

Reply:
Ok, but what if we need to support 2D shapes in future? 🙃

Reply:
Discussed offline. Agreed to leave as-is